core/common/src/format/LocalTimeFormat.kt

```diff
@@ -279,6 +279,25 @@ internal class LocalTimeFormat(override val actualFormat: CachedFormatStructure<
 }
 
 // these are constants so that the formats are not recreated every time they are used
+internal val ISO_TIME_BASIC by lazy {
+    LocalTimeFormat.build {
+        alternativeParsing({ char('t') }) { char('T') }
+        hour()
+        minute()
+        alternativeParsing({
+            // intentionally empty
+        }) {
+            optional {
+                second()
+            }
+            optional {
+                char('.')
+                secondFraction(1, 9)
+            }
+        }
```

Comment:
```suggestion
            optional {
                second()
                optional {
                    char('.')
                    secondFraction(1, 9)
                }
            }
        }
```
As the tests show, otherwise, `12:30:00.123` gets formatted as `T1230.123`, which is not valid: fractions of a second are only emitted together with the second value itself, even if it is zero.